### DISCUSSION TITLE
False incentive to the liquidator

### DISCUSSION THREAD

**Discussion by ebok21** [Cyfrin/foundry-full-course-cu#953] [General]
User A -> Deposits $150 ETH as collateral and mints $100 of DSC.
User B -> Deposits $250 ETH as collateral and mints $130 of DSC.

*Price of ETH tanks and now user A has defaulted*

User A -> Collateral value now worth $140 (less than the required health factor)
User B -> Liquidates User A by paying $130 worth of DSC (discounted price) and receives the collateral amount of User A.

Now having spent his DSC, User B has no way of getting his initial collateral of $250 ETH back and is stuck with much less ETH acquired by liquidating an undercollateralized user.

My question is, what even is the utility of these type of stablecoins??

**Comment by ebok21** [comment]
If @PatrickAlphaC could shed some light on this please

**Comment by 0xusmanf** [comment]
Good observation 👍🏻, this is a vulnerability of this protocol and has been pointed out [here](https://www.codehawks.com/finding/clm81m7y501jxw9rug0svjuto) during the audit. It will be fixed in the audited version of DSC.

**Reply by ebok21** [reply]
Wish I had completed the course before the contest, could've won a cool 1000 bucks😪

**Reply by PatrickAlphaC** [reply]
Be sure to sign up for CodeHawks :)
